File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int r=0,l=0,n=s.length();
        vector<int> count(92,0);
        int res=0;
        while (r<n){
            count[s[r]-' ']++;
            while(count[s[r]-' ']>1){
                count[s[l]-' ']--;
                l++;
            }
            res=max(res,r-l+1);
            r++;
        }
        return res;

    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-6.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res=[]
        n=len(nums)
        def backtrack(i,s,curr: List[int]):
            if i >=n or s> target:
                return 
            if s==target:
                res.append(curr.copy())
                return
            curr.append(nums[i])
            backtrack(i,s+nums[i],curr)
            curr.pop()
            backtrack(i+1,s,curr)
        backtrack(0,0,[])
        return res